iptables: skip NOTRACK rules deletion

jibi opened the pull request:
The Iptables version shipped with Cilium (1.8.4) does not fully support
the iptables-legacy - nft translation of NOTRACK rules, which in turn
causes the agent to crash whenever it tries to delete any old/backup
ruleset.

This commit introduces a workaround for this by ignoring the `-j
NOTRACK` and `-j CT --notrack` rules when deleting all rules belonging
to an old/backup ruleset.

This is not an issue (although it's not optimal) as eventually the agent
will just flush and remove all the Cilium managed Iptables chains.

Fixes: #20714

```release-note
Fix bug where Cilium would crash on startup with an error about being unable to delete iptables rules.
```

brb commented on pkg/datapath/iptables/iptables.go:
```diff
@@ -234,6 +234,18 @@ func (m *IptablesManager) removeCiliumRules(table string, prog iptablesInterface
+		// them here and let the agent remove them when it deletes the
+		// entire chain.
+		if strings.Contains(rule, "-j NOTRACK") ||
+			strings.Contains(rule, "-j CT --notrack") {
```
Doesn't the `-j CT --notrack` work on < 1.8.5?

jibi replied:
I think I tested restarting a 1.11 agent (which is using `-j CT --notrack`) and it failed to delete the old rules, but I might be mixing things up at this point, let me rerun the test

jibi replied:
You are right, no need to do anything with `-j CT --notrack` rules :+1:

jibi committed:
iptables: skip NOTRACK rules deletion

The Iptables version shipped with Cilium (1.8.4) does not fully support
the iptables-legacy - nft translation of NOTRACK rules, which in turn
causes the agent to crash whenever it tries to delete any old/backup
ruleset.

This commit introduces a workaround for this by ignoring the `-j
NOTRACK` and `-j CT --notrack` rules when deleting all rules belonging
to an old/backup ruleset.

This is not an issue (although it's not optimal) as eventually the agent
will just flush and remove all the Cilium managed Iptables chains.

Fixes: #20714
Signed-off-by: Gilberto Bertin <jibi@cilium.io>
